src/zmq/zmqpublishnotifier.cpp

```diff
@@ -206,3 +209,14 @@ bool CZMQPublishRawTransactionNotifier::NotifyTransaction(const CTransaction &tr
     ss << transaction;
     return SendMessage(MSG_RAWTX, &(*ss.begin()), ss.size());
 }
+
+bool CZMQPublishWalletTransactionNotifier::WalletTransactionChanged(CWallet* wallet, const uint256& hash, ChangeType status)
+{
+    std::ostringstream oss;
+    uint32_t name_size = htonl(wallet->GetName().size());
+    oss.write((const char*)&name_size, sizeof(name_size));
+    oss << wallet->GetName();
+    oss.write((const char*) hash.begin(), 32);
```

Comment:
Concept alternative: ZMQ supports [mulitpart messages](http://zguide.zeromq.org/php:chapter2#Multipart-Messages). Might make sense to split the wallet name, hash and reason over multiple frames rather than creating a _custom protocol_.

Reply:
I think we can't do that unless the payload goes after the sequence number - I mean if we want to be consistent with existing messages.

Reply:
Agree, that it would not be super consistent with the existing messages. However there was just never the need for more that one payload.

For discussion:
Currently we send a fixed 3 part multipart message. 
``` 
| topic | payload | sequence number |
```

I had an extension similar the following in mind when I read you PR description. 
``` 
| topic | wallet name | hash | (reason/change) | sequence number |
```

Reply:
Ah right, sequence is already a different part. Will update accordingly.